docs/USER_GROUP_API.md

```diff
@@ -199,6 +207,7 @@ error NoRecordFound()
 error BadService()
 error ServiceNotAvailable()
 error ConflictingRecordFound()
+error NonMatchingRecordFound()
```

Comment:
NonMatching sounds spurious to me. Maybe `NoMatching` is better??

Reply:
well, the point here is that a record *was* found by specified uid or name, but it's not matching the filter. NoMatchingRecordFound() would suggest that no match was found whatsoever, but NonMatchingRecordFound() means that a record was found but it was non-matching.

does that make sense? not sure how we could express that better.

i.e. the point i am trying to make here is that ther are 1st-class lookup keys (i.e. uid + name), and 2nd class filters (i.e. fuzzy name, uidmin/uidmax, disposition), and the first-class lookup keys always are more important and the filters are not, and we want to communicate that we managed to fulfill the 1st class lookup keys but not the 2nd class.

Reply:
Thanks. I reread the doc below, and actually it says so. But may be better to emphasize that.